service/src/test/java/bio/terra/workspace/service/resource/ValidationUtilsTest.java

```diff
@@ -239,4 +255,61 @@ public void validateBqDataTableName_invalidName_throwsException() {
         InvalidNameException.class, () -> ValidationUtils.validateBqDataTableName("00_お客様*"));
     assertThrows(InvalidNameException.class, () -> ValidationUtils.validateBqDataTableName(""));
   }
+
+  @Test
+  public void validateGitRepoUrl() {
+    validationUtils.validateGitRepoUri("https://github.com/path/to/project.git");
+    validationUtils.validateGitRepoUri("https://github.com/yuhuyoyo/testrepo.git");
+    validationUtils.validateGitRepoUri(
+        "git@github.com:DataBiosphere/terra-workspace-manager.git");
+    validationUtils.validateGitRepoUri("ssh://git@github.com/path/to/project.git");
+    validationUtils.validateGitRepoUri(
+        "https://username:password@github.com/username/repository.git");
+  }
+
+  @Test
+  public void validateGitRepoUrl_hostNameNotInAllowList_throwsException() {
+    assertThrows(
+        InvalidReferenceException.class,
+        () ->
+            validationUtils.validateGitRepoUri(
+                "ssh://git@github.com:DataBiosphere/terra-workspace-manager.git"));
+  }
+
+  @Test
+  public void validateGitRepoUrl_httpUrl_throwsException() {
+    assertThrows(
+        InvalidReferenceException.class,
+        () ->
+            validationUtils.validateGitRepoUri(
+                "http://github.com/DataBiosphere/terra-workspace-manager.git"));
+  }
+
+  @Test
+  public void test() throws URISyntaxException {
+    URI uri = new URI("ssh://github.com/project.git");
+
+    uri.getHost();
+  }
```

Comment:
does this test anything, or is it left over from development?

Reply:
oops nice catch